WooCommerce/Classes/ViewRelated/Orders/Refund/RefundSubmissionUseCase.swift

```diff
@@ -377,8 +377,16 @@ private extension RefundSubmissionUseCase {
     ///   - refund: the refund to submit.
     ///   - onCompletion: called when the submission completes.
     func submitRefundToSite(refund: Refund, onCompletion: @escaping (Result<Void, Error>) -> Void) {
-        let action = RefundAction.createRefund(siteID: details.order.siteID, orderID: details.order.orderID, refund: refund) { [weak self] _, error  in
+
+        let action = RefundAction.createRefund(siteID: details.order.siteID, orderID: details.order.orderID, refund: refund) { [weak self]
+            refundData, error  in
+
             guard let self = self else { return }
+
+            if let refundData = refundData {
+                // Workaround for https://github.com/woocommerce/woocommerce-ios/issues/6704. It can be removed when the API issue is fixed (Link TBD)
+                self.getLatestRefundData(refund: refundData) { [weak self] _ in }
```

Comment:
As explained below, perhaps we do not need the completion block (or even creating a function). But if the completion block remains, we do not need `[weak self]` here, because `self` is nowhere used.